tests/compiler/call-rest-err.ts

```diff
@@ -0,0 +1,14 @@
+function sum(...args: i32[]): i32 {
+  let sum = 0;
+  for (let i = 0, k = args.length; i < k; ++i) {
+    sum += args[i];
+  }
+  return sum;
+}
+
+function count<T>(...args: T[]): i32 {
+  return args.length;
+}
+
+sum('a', 'b'); // expect a type mismatch error on each argument
+count(1, 'a'); // expect a type mismatch error on the second argument
```

Comment:
Can you also test failure case when rest params placed not at the end? For example:
```ts
function wrongOrder<T>(...args: T[], extra: i32): i32 {
  return args.length + extra;
}
wrongOrder('a', 1); // expect a rest parameter must be last in a parameter list
```
Error:
```
A rest parameter must be last in a parameter list. (1014)
```

Reply:
Will do.

Reply:
Hmm, I don't see an expected error in stderr for `1014`